File: University/KamelBudanYaNabudanS1.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        try (Scanner scanner = new Scanner(System.in)) {

            int number = scanner.nextInt();
            int result = 1;

            for (int i = 2; i <= number/2; i++) {
                if( number % i == 0)
                    result += i;
            }
            if (result == number)
                System.out.print("YES");
            else
                System.out.print("NO");
        }
    }
}

// </Ctrl+Z>

File: University/KamelBudanYaNabudanS1.py
def main():
    number = int(input())
    result = 1

    for i in range(2, number // 2 + 1):
        if number % i == 0:
            result += i

    if result == number:
        print("YES")
    else:
        print("NO")

if __name__ == "__main__":
    main()
